File: baselib-java/src/main/java/com/kuky/baselib_java/baseMvpClass/BaseMvpPresenter.java
package com.kuky.baselib_java.baseMvpClass;

/**
 * @author Kuky
 */
public abstract class BaseMvpPresenter<V extends BaseMvpViewImpl> {
    protected V mView;

    protected void attach(V view) {
        this.mView = view;
    }

    protected void detach() {
        this.mView = null;
    }

    protected void onResume() {

    }

    protected void onStop() {

    }

    protected void onPause() {

    }
}

File: baselib/src/main/java/com/kuky/baselib/baseMvpClass/BaseMvpPresenter.kt
package com.kuky.baselib.baseMvpClass

/**
 * @author Kuky
 */
abstract class BaseMvpPresenter<V : BaseMvpViewImpl> {
    protected var mView: V? = null

    fun attachView(view: V) {
        this.mView = view
    }

    fun detachView() {
        this.mView = null
    }

    fun onResume() {

    }
}